src/report.ts

```diff
@@ -4,9 +4,17 @@ import * as models from 'powerbi-models';
 import * as wpmp from 'window-post-message-proxy';
 import * as hpm from 'http-post-message';
 import * as utils from './util';
+import { IFilterable } from './ifilterable';
+import { IPage, Page } from './page';
 
-export class Report extends embed.Embed {
-    static allowedEvents = ["dataSelected", "filterAdded", "filterUpdated", "filterRemoved", "pageChanged", "error"];
+export interface IReport {
+    iframe: HTMLIFrameElement;
+    service: service.IService;
+    config: embed.IInternalEmbedConfiguration
+}
+
+export class Report extends embed.Embed implements IReport, IFilterable {
+    static allowedEvents = ["dataSelected", "filtersApplied", "pageChanged", "error"];
```

Comment:
Yea, you had commented on this in previous review.  The loaded event is for all embeds so it's in the generic embed class.